caikit/interfaces/nlp/data_model/classification.py

```diff
@@ -0,0 +1,136 @@
+class Classification(DataObjectBase):
+    """A single classification prediction."""
+
+    label: Annotated[str, FieldNumber(1)]  # Predicted relevant class name
+    score: Annotated[
+        float, FieldNumber(2)
+    ]  # The confidence-like score of this prediction in [0, 1]
+
+
+@dataobject(package=NLP_PACKAGE)
+class ClassificationResult(DataObjectBase):
+    """Classification result generated from a text and consisting multiple classes."""
+
+    results: Annotated[
+        List[Classification], FieldNumber(1)
+    ]  # List of classifications for a text
+
+
+# NOTE: This is meant to align with the HuggingFace token classification task:
+# https://huggingface.co/docs/transformers/tasks/token_classification#inference
+# The field `word` does not necessarily correspond to a single "word",
+# and `entity` may not always be applicable beyond "entity" in the NER
+# (named entity recognition) sense
+@dataobject(package=NLP_PACKAGE)
+class TokenClassification(DataObjectBase):
+    """A single token classification prediction."""
+
+    start: Annotated[int, FieldNumber(1)]  # Beginning offset of the token
+    end: Annotated[int, FieldNumber(2)]  # Ending offset of the token
+    word: Annotated[str, FieldNumber(3)]  # Text referenced by this token
+    entity: Annotated[
+        str, FieldNumber(4)
+    ]  # Predicted relevant class name for the token
+    entity_group: Annotated[str, FieldNumber(5)]  # Aggregate label, if applicable
+    score: Annotated[
+        float, FieldNumber(6)
+    ]  # The confidence-like score of this classification prediction in [0, 1]
+    token_count: Annotated[
+        Optional[int], FieldNumber(7)
+    ]  # Length of tokens in the text
+
+
+@dataobject(package=NLP_PACKAGE)
+class TokenClassificationResult(DataObjectBase):
+    """Token classification result generated from a text and consisting multiple classes."""
+
+    results: Annotated[List[TokenClassification], FieldNumber(1)]
+
+
+@dataobject(package=NLP_PACKAGE)
+class TokenClassificationStreamResult(TokenClassificationResult):
+    """
+    Streaming token classification result that indicates up to where in stream is processed.
+    """
+
+    processed_index: Annotated[
+        int, FieldNumber(2)
+    ]  # Result index up to which text is processed
+
+
+@dataobject(package=NLP_PACKAGE)
+class ClassifiedGeneratedTextResult(DataObjectBase):
+    """Classification result on text produced by a text generation model, contains
+    information from the original text generation output as well as the result of
+    classification on the generated text.
+    """
+
+    text: Annotated[str, FieldNumber(1)]  # The generated text
+    token_classification_results: Annotated[
+        Optional[List[TokenClassification]], FieldNumber(2)
+    ]  # Token classification results for this generated text
+    finish_reason: Annotated[
+        Optional[FinishReason], FieldNumber(3)
+    ]  # Reason as to why text generation stopped
+    token_count: Annotated[
```

Comment:
this seems to be duplicate as `token_count` also exist in `token_classification_results` ?